plugin/trino-hive/src/main/java/io/trino/plugin/hive/PartitionsSystemTableProvider.java

```diff
@@ -68,11 +83,23 @@ public PartitionsSystemTableProvider(HivePartitionManager partitionManager)
         }
 
         SchemaTableName sourceTableName = PARTITIONS.getSourceTableName(tableName);
-        HiveTableHandle sourceTableHandle = metadata.getTableHandle(session, sourceTableName);
-
-        if (sourceTableHandle == null) {
+        Table sourceTable = metadata.getMetastore()
+                .getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName())
+                .orElse(null);
+        if (sourceTable == null || isIcebergTable(sourceTable)) {
             return Optional.empty();
         }
+        if (isDeltaLakeTable(sourceTable)) {
```

Comment:
I would add a new commit to change the behaviour for querying hive system tables on delta lake tables, in the same fashion as for iceberg tables, to not throw a unsupported exception, but rather to return `Optional.empty()`.
This would provide a consistent approach for dealing with querying on system tables for non-hive tables.

Reply:
Let's do this as a prep commit in this PR. Otherwise Iceberg handling being added here is glaringly inconsistent with Delta Lake case, for no apparent reason.

Reply:
Yes, a prep commit makes much more sense to improve the readability of the changes performed while working on this issue.